smpplib/client.py

```diff
@@ -61,12 +62,14 @@ class Client(object):
     vendor = None
     _socket = None
     sequence_generator = None
-
-    def __init__(self, host, port, timeout=5, sequence_generator=None, logger_name=None):
+    
+    def __init__(self, host, port, timeout=5, sequence_generator=None, max_outstanding_operations=None, logger_name=None):
         self.host = host
         self.port = int(port)
         self._socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
         self.timeout = timeout
+        self.outstanding_operations = 0
```

Comment:
I'm a bit afraid of the counter getting out of sync for requests/responses. Would it be smarter to keep a `set` of PDU sequence IDs? So, when you got a request PDU then you add its sequence ID to the set. For a response PDU you remove its sequence ID from the set. And to check you test length of the set. Then we're more safe in terms of lost packets and/or retries

By the way, I guess that `unbind` and disconnect handlers should reset the current outstanding operations counter/set. Otherwise, we may never get response for some unconfirmed PDUs

Reply:
I think using a set is a good idea yes.

Regarding unbind/disconnect, it shouldn't reset it. Example:
I connect, send 10 submit_sm and then disconnect. When I reconnect 5 minutes later, the SMSC will actually send me 10 submit_sm_resp that were outstanding, using the same sequence numbers that were used to send the submit_sm.
We don't reset the sequence number on unbind/disconnect by default either, do we ?

Nevertheless it wouldn't be failproof in case the script gets killed or something so...